### PR TITLE
Pervasives deprecated in 408

### PR DESCRIPTION
**Pull request by anuragsoni**
The `Pervasives`  module is deprecated in ocaml 4.08 -> https://github.com/ocaml/ocaml/pull/1605

This pull request uses dune's configurator to use the `Stdlib` module in ocaml versions 4.08 and above, and keeps the old pervasives module in versions older than that.

### CODE REVIEW

**Comment by pmetzger** [comment]
Given that this is several updates at once, could you edit the description to tell us what you've done? That makes it easier to evaluate.

**Comment by anuragsoni** [comment]
@pmetzger Thanks for the feedback. I updated the description. Although to be fair its just one update to avoid using the Pervasives module in versions 4.08 and up.

**Comment by anuragsoni** [comment]
On a sidenote, it might be nice if one of the admins can convert `3.x` to `master` or at-least make the development branch be the default on github. It'll make it easier to discover.

**Comment by pmetzger** [comment]
> On a sidenote, it might be nice if one of the admins can convert 3.x to master or make the development branch be the default on github. It'll make it easier to discover.

@anuragsoni So the problem is that the maintainer has sort of dropped out. If you would like to take over, or even just help, I can give you commit access to this repository.

**Comment by anuragsoni** [comment]
@pmetzger I understand. 

Unfortunately I won't have time to dedicate for full-time maintainer's role just yet (atleast for a couple of months). I'm not sure that a calendar/date/time library is something that should be maintained who hasn't done much work in such libraries. What i can do though is experiment with calenderlib in my own fork, try and see if i can improve documentation to improve my own knowledge (and maybe help others looking at this). I'll make pull requests with changes are relevant and reach out once I have more time at hand to help with maintainer duties.

**Comment by pmetzger** [comment]
Even if you can't guarantee much long term, would you like access to fix the branch vs. trunk issue and to fix the build for 4.08?

**Comment by anuragsoni** [comment]
Sure i wouldn't mind updating the branches.

**Comment by XVilka** [comment]
Would be good to carve the new release after that: https://github.com/ocaml-community/calendar/issues/23

**Comment by jeromesimeon** [comment]
> Would be good to carve the new release after that: #23

Does anyone has a sense of how this release will differ from the last published version on opam?

**Comment by anuragsoni** [comment]
@jeromesimeon As far as i can tell the differences are mostly on the tooling side. The build system was switched to dune. I don't think there has been any change in the implementation.

**Comment by loxs** [comment]
@jeromesimeon @anuragsoni probably the only breaking change is this. Although "correct" IMO, it might break some legacy systems: https://github.com/ocaml-community/calendar/issues/14

**Comment by jeromesimeon** [comment]
A quick look at the commit log, I noted those:

# Fixes

- Week of year was off by one in some cases (https://github.com/ocaml-community/calendar/issues/14)
- To the precision used in equality for dates (https://github.com/ocaml-community/calendar/commit/0e073eeae4ee8cb811e186755aaf959ba21e221e)
- Remove incorrect time zone bound check (https://github.com/ocaml-community/calendar/commit/49ced035269728bd7283061b31b6b60de1901596)

# Build

- Support for OCaml ?? (>= 4.08.0 ?)
- Build system switched to dune
- Switch from Str to Re, for browser compatibility

**Comment by pmetzger** [comment]
@anuragsoni You have write permission now. I'd suggest that you mostly use pull requests to get things reviewed, but you can commit on your own after review. Feel free to (very carefully, and after testing locally!) fix the branch situation on your own.

**Comment by anuragsoni** [comment]
@pmetzger Thanks. @Drup Since you are mentioned in the codeowners file, if you think this patch looks okay i can merge this.

**Comment by Drup** [comment]
@anuragsoni Why didn't you used `stdlib-shim` ? It's literally made for that purpose ?

**Comment by Drup** [comment]
Actually, looking at the code, those are not even really useful, the only thing you want is `Int.compare`. You don't need dune configurator just to get that.

**Comment by anuragsoni** [comment]
Thanks for pointing out stdlib-shims. I wasn't aware of it. But like you said since the only thing we need at the moment is Int.compare, i'll remove the dune configurator. I should be able to get rid of using `Pervasives` without that too. I'll update the PR later today when i have some time.

**Comment by anuragsoni** [comment]
The PR is now updated to no longer need dune configurator

File: src/date.ml [hunk 1]
```diff
@@ -75,8 +75,16 @@ let hash = Utils.Int.hash
...
+let lt ((y1, m1, d1) : int * int * int) ((y2, m2, d2) : int * int * int) =
+  let cmp = match compare y1 y2 with
+  | 0 -> begin
+      match compare m1 m2 with
+      | 0 -> compare d1 d2
+      | i -> i
+    end
+  | i -> i
+  in
+  cmp < 0
```
Comment (Drup) [inline src/date.ml]: Shorter and nicer:
```ocaml
let cmp_int (a: int) b = compare a b
let (<?>) c (ord,x,y) =
  if c = 0 then ord x y else c
let lt ((y1, m1, d1) : int * int * int) ((y2, m2, d2) : int * int * int) =
  cmp_int y1 y2 <?> (cmp_int, m1, m2) <?> (cmp_int, d1, d2)
```

Comment (Drup) [reply src/date.ml]: Actually, since you're on it, you might as well introduce properly efficient comparisons everywhere :)

Comment (anuragsoni) [reply src/date.ml]: The infix operator indeed makes things very readable here. Thanks!

For the other comparisons i believe the ones i saw so far were already using the integer compare from the `Int` module inside utils (`let compare (a: int) b = compare a b`) 

I'll take a second pass to find places i missed. I will be on vacation for a week so i won't be able to update the pull request for a while.

**Comment by jeromesimeon** [comment]
Hopeful ping to any of the maintainers: This PR looks done to me?

**Comment by anuragsoni** [comment]
@jeromesimeon At this point i don't have too much motivation to continue with this PR so i'll be happy if someone else can take it through the rest of the way. At the time of the initial PR my goal was just to help out the current maintainer with something that i was hoping would be a small easy to review PR. 

I think your best bet will be to make a post on discourse https://discuss.ocaml.org to see if there is someone else who will be willing to step in as a maintainer.

**Comment by jeromesimeon** [comment]
> @jeromesimeon At this point i don't have too much motivation to continue with this PR so i'll be happy if someone else can take it through the rest of the way. At the time of the initial PR my goal was just to help out the current maintainer with something that i was hoping would be a small easy to review PR.
> 
> I think your best bet will be to make a post on discourse https://discuss.ocaml.org to see if there is someone else who will be willing to step in as a maintainer.

@anuragsoni Thanks for the response. That makes sense to me (and as far as I'm concerned, that PR looks done! just no one is around to merge it?). I'll try and post on discourse when I can find a moment.

**Comment by pmetzger** [comment]
I can merge something if there's consensus to merge it. That said, I already gave @anuragsoni commit privileges.

**Comment by anuragsoni** [comment]
@pmetzger I know i had the commit privileges but for projects where i'm not a maintainer i prefer when someone else goes over the change and there is a consensus to merge. I don't feel too comfortable merging my own changes for projects that I don't maintain.

**Pull request merged**